src/Utility/Snapshots.lua

```diff
@@ -163,9 +151,8 @@ function Utility.CreatePlayersSnapshot(Time: number)
 			continue
 		end
 
-		local RigType = Character.Humanoid.RigType.Value :: number
-		local PARTS = CHARACTER_DATA[RigType].Parts
-		local PARTS_SIZE  = CHARACTER_DATA[RigType].Size
+		local RigType = Character.Humanoid.RigType :: Enum.HumanoidRigType
+		local PartNames, PartSizes = PARTS[RigType].Names, PARTS[RigType].Sizes
```

Comment:
`PartSizes` can be removed here, it's only use is for the length check but that can be accomplished using PartNames since they have the same length